src/redux/books/books.js

```diff
@@ -0,0 +1,55 @@
+import { v4 } from 'uuid';
+
+const ADD = 'bookstore/books/ADD';
+const REMOVE = 'bookstore/books/REMOVE';
+
+// Actions
+const addBook = (book) => ({
+  type: ADD,
+  book,
+});
+
+const removeBook = (id) => ({
+  type: REMOVE,
+  id,
+});
+
+// Reducer
+const bookReducer = (state = [
+  {
+    id: v4(),
+    title: 'The Pearl of a great price',
+    author: 'Joseph Smith',
+    genre: 'Gospel',
+  },
+  {
+    id: v4(),
+    title: 'The Prince',
+    author: 'Nicolo Makiaveli',
+    genre: 'philosophy',
+  },
+  {
+    id: v4(),
+    title: 'Price of my karma',
+    author: 'Patrick Mukul',
+    genre: 'philosophy',
+  },
+  {
+    id: v4(),
+    title: 'Metaphore',
+    author: 'Pitagore',
+    genre: 'historical',
+  },
+], action = {}) => {
+  switch (action.type) {
+    case ADD:
+      return [...state, action.book];
+    case REMOVE:
+      return [...state.filter((book) => book.id !== action.id)];
+    default:
+      return state;
+  }
+};
+
+export { addBook, removeBook };
+export default bookReducer;
```

Comment:
- [ OPTIONAL ] Great job so far @Patfarmurs. Your app works great, I will suggest that you make use of `createSlice` from `@reduxjs/toolkit` as it provides several advantages over the `traditional switch statement syntax `used to define a reducer function. :+1: 
- Here are some reasons why using `createSlice` is a better approach:

1. Less Boilerplate: `createSlice` reduces the amount of boilerplate code needed to create a reducer function by handling the creation of the action creators and reducer logic in one step. This makes it quicker and easier to write and maintain code. :100: 

2. Immutable Updates: `createSlice` uses `Immer` internally to handle immutable updates to the state. This means that you can write code that looks like you're directly modifying the state, but it creates a new immutable copy of the state. This helps avoid accidental mutations of state and makes it easier to reason about the application's state. This is something you will need in future projects to ensure that the state of your app is immutable :+1: 

Overall, using `createSlice` from `@reduxjs/toolkit `can lead to more concise, maintainable, and performant code. :100: You can learn more about it [here](https://redux-toolkit.js.org/api/createSlice) :wink:

Reply:
Thanks for your suggestion @NahnahAJ, I'll learn more about createSlice and make use of it in my future projects.

Reply:
You're welcome @Patfarmurs :wink: